inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,220 @@
-                "value": form_fields['conf_name']
-            })
-            data['hidden_notes'].append({
-                'value': form_fields['nonpublic_note']
-            })
-        else:
-            data.setdefault("hidden_notes", []).append({
-                "value": form_fields['conf_name']
-            })
-        data['collections'].extend([{'primary': "ConferencePaper"}])
-
-    # ============================
-    # Page number
-    # ============================
-    if 'page_nr' not in data:
-        first_publication_info = data.get('publication_info', [{}])[0]
-
-        page_start = first_publication_info.get('page_start')
-        page_end = first_publication_info.get('page_end')
-
-        if page_start and page_end:
+        year = int(year)
+    except (TypeError, ValueError):
+        year = None
+
+    # journal title
+    if form_fields.get('journal_title'):
+        hits = JournalsSearch().query(
+            'match', title_variants__title__lowercased=form_fields['journal_title']).execute()
+
+        if hits:
             try:
-                data['page_nr'] = int(page_end) - int(page_start) + 1
-            except (TypeError, ValueError):
+                form_fields['journal_title'] = hits[0].short_titles[0].title
+            except (AttributeError, IndexError):
                 pass
 
-    # ============================
-    # Language
-    # ============================
-    if form_fields.get('language') == 'oth':
-        if form_fields.get("other_language"):
-            data["languages"] = [form_fields["other_language"]]
-
-    # ==========
-    # Owner Info
-    # ==========
-    # TODO Make sure we are getting the email correctly
-    userid = obj.id_user
+    # journal location
+    page_range = form_fields.get('page_range_article_id')
+
+    artid = None
+    page_end = None
+    page_start = None
+    if page_range:
+        page_start, page_end, artid = split_page_artid(page_range)
+
+    builder.add_publication_info(
+        year=year,
+        cnum=form_fields.get('conference_id'),
+        journal_issue=form_fields.get('issue'),
+        journal_title=form_fields.get('journal_title'),
+        journal_volume=form_fields.get('volume'),
+        page_start=page_start,
+        page_end=page_end,
+        artid=artid
+    )
+
+    if form_fields.get('preprint_created'):
+        builder.add_imprint_date(form_fields.get('preprint_created'))
+
+    builder.add_thesis(
+        defense_date=form_fields.get('defense_date'),
+        degree_type=form_fields.get('degree_type'),
+        institution=form_fields.get('institution'),
+        date=form_fields.get('thesis_date')
+    )
+    if form_fields.get('experiment'):
+        builder.add_accelerator_experiment(experiment=form_fields.get('experiment'))
+
+    language = _language_mapping(
+        form_fields.get('language'),
+        form_fields.get('other_language')
+    )
+    if language and len(language) == 2:
```

Comment:
Actually now there is an issue with the language. I https://github.com/inspirehep/inspire-next/issues/1915